plugins/opusCodec/src/OpusEncoder.cpp

```diff
@@ -0,0 +1,273 @@
+    return returnValue;
+}
+
+void AthenaOpusEncoder::setComplexity(int complexity) {
+    assert(_encoder);
+    int returnValue = opus_encoder_ctl(_encoder, OPUS_SET_COMPLEXITY(complexity));
+
+    if (returnValue != OPUS_OK) {
+        qCWarning(encoder) << "Error when setting complexity to " << complexity << ": " << errorToString(returnValue);
+    }
+}
+
+int AthenaOpusEncoder::getBitrate() const {
+    assert(_encoder);
+    int returnValue;
+    opus_encoder_ctl(_encoder, OPUS_GET_BITRATE(&returnValue));
+    return returnValue;
+}
+
+void AthenaOpusEncoder::setBitrate(int bitrate) {
+    assert(_encoder);
+    int errorCode = opus_encoder_ctl(_encoder, OPUS_SET_BITRATE(bitrate));
+
+    if (errorCode != OPUS_OK) {
+        qCWarning(encoder) << "Error when setting bitrate to " << bitrate << ": " << errorToString(errorCode);
+    }
+}
+
+int AthenaOpusEncoder::getVBR() const {
+    assert(_encoder);
+    int returnValue;
+    opus_encoder_ctl(_encoder, OPUS_GET_VBR(&returnValue));
+    return returnValue;
+}
+
+void AthenaOpusEncoder::setVBR(int vbr) {
+    assert(_encoder);
+    int errorCode = opus_encoder_ctl(_encoder, OPUS_SET_VBR(vbr));
+
+    if (errorCode != OPUS_OK) {
+        qCWarning(encoder) << "Error when setting VBR to " << vbr << ": " << errorToString(errorCode);
+    }
+}
+
+int AthenaOpusEncoder::getVBRConstraint() const {
+    assert(_encoder);
+    int returnValue;
+    opus_encoder_ctl(_encoder, OPUS_GET_VBR_CONSTRAINT(&returnValue));
+    return returnValue;
+}
+
+void AthenaOpusEncoder::setVBRConstraint(int vbr_const) {
+    assert(_encoder);
+    int errorCode = opus_encoder_ctl(_encoder, OPUS_SET_VBR_CONSTRAINT(vbr_const));
+    
+    if (errorCode != OPUS_OK) {
+        qCWarning(encoder) << "Error when setting VBR constraint to " << vbr_const << ": " << errorToString(errorCode);
+    }
+}
+
+int AthenaOpusEncoder::getMaxBandwidth() const {
+    assert(_encoder);
+    int returnValue;
+    opus_encoder_ctl(_encoder, OPUS_GET_MAX_BANDWIDTH(&returnValue));
+    return returnValue;
+}
+
+void AthenaOpusEncoder::setMaxBandwidth(int maxbw) {
+    assert(_encoder);
+    int errorCode = opus_encoder_ctl(_encoder, OPUS_SET_MAX_BANDWIDTH(maxbw));
+    
+    if (errorCode != OPUS_OK) {
+        qCWarning(encoder) << "Error when setting max bandwidth to " << maxbw << ": " << errorToString(errorCode);
+    }
+}
+
+int AthenaOpusEncoder::getBandwidth() const {
+    assert(_encoder);
+    int bandwidth;
+    opus_encoder_ctl(_encoder, OPUS_GET_BANDWIDTH(&bandwidth));
+    return bandwidth;
+}
+
+void AthenaOpusEncoder::setBandwidth(int bandwidth) {
+    assert(_encoder);
+    int errorCode = opus_encoder_ctl(_encoder, OPUS_SET_BANDWIDTH(bandwidth));
+
+    if (errorCode != OPUS_OK) {
+        qCWarning(encoder) << "Error when setting bandwidth to " << bandwidth << ": " << errorToString(errorCode);
+    }
+}
+
+int AthenaOpusEncoder::getSignal() const {
+    assert(_encoder);
+    int signal;
```

Comment:
does this still compile? signal looks blue on my screen, it could be a reserved keyboard

Reply:
My keyboard is indeed private. As for the `signal` variable, it doesn't appear as a reserved word in my IDE, and it compiles just fine.

Reply:
Ah ok, its probably highlighted in all c++ on github just in case they're using the std namespace: https://en.cppreference.com/w/cpp/utility/program/signal